driver-core/src/main/com/mongodb/client/model/Aggregates.java

```diff
@@ -681,127 +664,68 @@ public static Bson search(final SearchOperator operator) {
      *
      * @param operator A search operator.
      * @param options Optional {@code $search} pipeline stage fields.
-     * Specifying {@code null} is equivalent to either specifying {@link SearchOptions#defaultSearchOptions()}
-     * or calling {@link #search(SearchOperator)}.
      * @return The {@code $search} pipeline stage.
      *
      * @mongodb.atlas.manual atlas-search/query-syntax/#-search $search
      * @mongodb.atlas.manual atlas-search/operators-and-collectors/#operators Search operators
      * @mongodb.atlas.manual atlas-search/scoring/ Scoring
      * @since 4.7
      */
-    public static Bson search(final SearchOperator operator, @Nullable final SearchOptions options) {
+    public static Bson search(final SearchOperator operator, final SearchOptions... options) {
```

Comment:
This feels hacky and less clear, would you ever expect more that one options here?

Reply:
This is not to allow specifying `SearchOptions` more than once, but to allow specifying it 0 or 1 times. However, the implementation still processes multiple `SearchOptions` in the unlikely case that they are specified.

Reply:
We'd still have to handle it in Scala and as an option type - which is the appropriate idiomatic scala type in this scenario.

I just don't think its very clear that there should be 0 or 1 options. I definitely think it should be an error if multiple options are supplied. 

I think here I'm of the opinion just because you can doesn't mean you should. I don't see a benefit over two explicit methods from the perspective of a user of the API.  I also think the overloads aren't very costly and its us that should be baring that cost as developers and not adding a cognitive cost to users.

Reply:
> We'd still have to handle it in Scala and as an option type

Scala has varargs, and I am not sure that wrapping a vararg in `Option` is an idiomatic approach.

As for the cost of overloads, I know, I am alone in the camp of those who thinks they are bad because of the duplication in Scala API, and because of the duplicated documentation.

What is your opinion regarding the other thing done in the PR: replacing `T single` with `T first, T... more`?

Reply:
> Scala has varargs, and I am not sure that wrapping a vararg in Option is an idiomatic approach.
The idiomatic approach is to look at the correct type for SearchOptions eg: which is expected type: `Seq[SearchOptions]` or `Option[SearchOptions]`

Reply:
> What is your opinion regarding the other thing done in the PR: replacing `T single` with `T first, T... more`?

That pattern is more comment and is clearer in the requirement of being 1 or more.

Reply:
Right, if the type is `Seq[T]`/`Option[T]`, but what if the method uses neither as the parameter type, and the parameter is expressed as `method(args: T*)` instead? If no values are provided, i.e., the method is called as `method()`, then `args` is empty (not `null`) in Scala (in Java it is `null`). So not specifying any values of a vararg parameter in Scala does not introduce `null`s. If we say that expressing a parameter that may have no values as varargs is not idiomatic in either Java or Scala, we are effectively saying that a varargs parameter must always have at least one value. However, that begs the question: why does the syntax of both languages allow specifying 0 vararg values?

Reply:
> However, that begs the question: why does the syntax of both languages allow specifying 0 vararg values?

In Scala `args: T*` is syntactic sugar for `Seq[T]` - which I suppose is the same as Java's varargs being syntactic sugar `String.... args` and `String[] args`. 

So I suppose as arrays can be empty it defaults to that :man_shrugging: ?

Reply:
In the scenario where its 0 or 1 - I think we'd need to clarify:

* That more than 1 is an error.
* That it is an optional parameter (just java doc).
* That by introducing yet another way of doing things we aren't adding confusion to the API but improving it.

Reply:
Jeff expressed the opinion similar to your original impression - unusual/don't like. I will undo the changes related to using `T...` for representing optional paramaters.

Reply:
Rolled back the changes related to expressing optional parameters as varargs. If the team changes its mind regarding this approach before the release of 4.7.0, we can do that part in a separate PR. For now, without that change, I hope that the PR is no longer controversial and has a potential to be merged.

Done.